frontend/javascripts/oxalis/model/reducers/flycam_reducer.ts

```diff
@@ -267,11 +271,34 @@ function FlycamReducer(state: OxalisState, action: Action): OxalisState {
         return { name, value: fallbackValue };
       });
 
-      return update(state, {
+      const updatedFlycamState = update(state, {
         flycam: {
           additionalCoordinates: { $set: values },
         },
       });
+
+      const visibleSegmentationLayer = getVisibleSegmentationLayer(state);
+      const additionalCoordinateString = getAdditionalCoordinatesAsString(values);
+      if (
+        visibleSegmentationLayer == null ||
+        state.localSegmentationData[visibleSegmentationLayer.name].meshes[
+          additionalCoordinateString
+        ] != null
+      ) {
+        return updatedFlycamState;
+      }
+
+      const updatedLocalSegmentationState = update(updatedFlycamState, {
+        localSegmentationData: {
+          [visibleSegmentationLayer.name]: {
+            meshes: {
+              [additionalCoordinateString]: { $set: [] },
+            },
+          },
+        },
+      });
```

Comment:
I assume that this was added so that access to `localSegmentationData[layerName].meshes[coords][...]` does not fail due to missing initialization. I think, it would be better to not have to rely on this (but instead to change the code that reads from the store so that is robust over missing keys).
You could change the type of the nested localSegmentationData object to something like:

```ts
localSegmentationData: Record<
  string,
  Record<
    string,
    Record<...> | undefined
  > | undefined
>
```

That way, each look up would need to check for undefined (which you already do at most cases, anyway) and typescript should enforce this.

Maybe you could give this a try. If it turns out to be too tedious, I'm also fine with keeping it as is for now :)

Reply:
In this code I tried to make sure that entries for new additional coordinates are added. I moved this code so that new entries are only created if actual meshes are present. Other than that I also changed the types, which really was a little tedious, but the code seems to handle errors better now, so I hope it was worth it. :)